src/server/session.ts

```diff
@@ -25,15 +25,17 @@ namespace ts.server {
         return ((1e9 * seconds) + nanoseconds) / 1000000.0;
     }
 
-    function shouldSkipSemanticCheck(project: Project) {
-        if (project.projectKind === ProjectKind.Inferred || project.projectKind === ProjectKind.External) {
-            return project.isJsOnlyProject();
-        }
-        else {
-            // For configured projects, require that skipLibCheck be set also
-            const options = project.getCompilerOptions();
-            return options.skipLibCheck && !options.checkJs && project.isJsOnlyProject();
+    function shouldSkipSemanticCheck(project: Project, file: NormalizedPath) {
+        // For inferred (e.g. miscellaneous context in VS) and external projects (e.g. VS .csproj project) with only JS files
+        // semantic errors in .d.ts files (e.g. ones added by automatic type acquisition) are not interesting.
+        // We want to avoid the cost of querying for these errors all together, even if 'skipLibCheck' is not set.
+        // We still want to check .js files (e.g. '// @ts-check' or '--checkJs' is set).

```

Comment:
Maybe a comment to that effect in this function then? It's confusing to have a function called `shouldSkipSemanticCheck` that takes a file, that doesn't actually check if it should skip the semantic check for that file 😕